app/controllers/application_controller.rb

```diff
@@ -1,2 +1,12 @@
 class ApplicationController < ActionController::Base
+  def is_loggedin
+    if current_user.present?
+      redirect_to root_path
+      return
+    end
+  end
+
+  def current_user
+    session[:customer_id].present? ? Customer.find_by(id: session[:customer_id]) : nil
+  end
```

Comment:
1. Should we make these methods private?
2. I think this logic can if refactored
```suggestion
  def authenticate_customer!
    redirect_to root_path and return unless current_customer.present?
  end

  def current_customer
    Customer.find_by(id: session[:customer_id]) if session[:customer_id].present?
  end
```